src/include/network.h

```diff
@@ -57,6 +57,16 @@ pgagroal_bind(const char* hostname, int port, int** fds, int* length);
 int
 pgagroal_bind_unix_socket(const char* directory, const char* file, int* fd);
 
+/**
+ * Bind sockets for a host
+ * @param hostname The host name
+ * @param port The port number
+ * @param fd The resulting descriptor
+ * @return 0 upon success, otherwise 1
+ */
+int
+pgagroal_vault_bind(const char* hostname, int port, int** fd);
```

Comment:
The shared library provides functions like `pgagroal_bind` and `pgagroal_connect` which uses some fields of `struct configuration` internally which are not there in `struct vault_configuration`, also these functions assumes logging to be started while in vault I was not using logging anywhere.

Reply:
See if

```
struct configuration
{
 ...
}

struct main_configuration
{
   struct configuration config;
  ...
}

struct vault_configuration
{
   struct configuration config;
  ...
}
```

will work - otherwise we will go your current route. At least it is something to think about...